src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -2794,6 +2803,8 @@ internal override StrongNameKeys StrongNameKeys
                     filterOpt: filterOpt,
                     cancellationToken: cancellationToken);
 
+                GenerateModuleInitializer(moduleBeingBuilt, methodBodyDiagnosticBag);
```

Comment:
>GenerateModuleInitializer [](start = 16, length = 25)

We probably shouldn't call this method when ```hasDeclarationErrors``` is true.

Reply:
And also when ```CommonCompiler.HasUnsuppressableErrors(methodBodyDiagnosticBag)```

---
In reply to: [415088304](https://github.com/dotnet/roslyn/pull/43281#discussion_r415088304) [](ancestors = 415088304)

Reply:
Should I look for a way to write a test to enforce each of these things?

Reply:
> Should I look for a way to write a test to enforce each of these things?

I don't think it is worth doing.

---
In reply to: [415121568](https://github.com/dotnet/roslyn/pull/43281#discussion_r415121568) [](ancestors = 415121568)

Reply:
So sorry, I thought I remembered doing this.

Reply:
Done by Rikki in https://github.com/dotnet/roslyn/pull/43964.